apis/object/v1alpha1/types.go

```diff
@@ -124,6 +124,14 @@ type ObjectSpec struct {
 	ManagementPolicy `json:"managementPolicy,omitempty"`
 	References       []Reference `json:"references,omitempty"`
 	Readiness        Readiness   `json:"readiness,omitempty"`
+	// Watch enables watching the referenced or managed kubernetes resources.
+	//
+	// THIS IS AN ALPHA FIELD. Do not use it in production. It is not honored
+	// unless the relevant Crossplane feature flag is enabled, and may be
+	// changed or removed without notice.
+	// +optional
+	// +kubebuilder:default=false
+	Watch *bool `json:"watch,omitempty"`
```

Comment:
Initially, I didn't have this field in the spec and it was an all-in feature (similar to Crossplane realtime compositions). 

However, after my experiments and tests with our internal product leveraging provider-kubernetes, I realized, I want to enable watching only a very limited subset of the resources not all of them. For most of the resources, it is totally fine if provider-kubernetes detects drift after poll interval (e.g. 10 mins by default) since we don't really expect any drift under normal circumstances. Similar to how a helm release deploys some resources and have no drift detection/active reconciliation and this is being fine for most of the use cases. So, I ended up having this as optional spec to start with, in addition to having this as an alpha feature.

Reply:
Why a pointer?

Reply:
Just following the convention of defining optional fields as pointers, not aware of any practical benefits/differences TBH.

Reply:
for basic types with omitempty this is overcomplicated everything, like risking panics.